source/parser/lib.civet

```diff
@@ -1268,8 +1263,34 @@ async function processProgramAsync(root: BlockStatement): Promise<void>
   { expressions: statements } := root
   await processComptime(statements)
 
+function processRepl(root: BlockStatement, rootIIFE: ASTNode): void
+  topBlock :=
+    gatherRecursive(rootIIFE!, .type is "BlockStatement")[0]
+  i .= 0
+  // Hoist top-level declarations and all var declarations
+  for each decl of gatherRecursiveWithinFunction topBlock, .type is "Declaration"
+    if decl.parent is topBlock or decl.decl is "var"
```

Comment:
We should add a test to make sure we don't hoist `vars` that are inside function scopes..